examples/js/loaders/GLTFLoader.js

```diff
@@ -1547,59 +1528,74 @@ THREE.GLTFLoader = ( function () {
 
 	}
 
-	function isArrayEqual( a, b ) {
+	function createGeometryKey( primitiveDef ) {
 
-		if ( a.length !== b.length ) return false;
+		var dracoExtension = primitiveDef.extensions && primitiveDef.extensions[ EXTENSIONS.KHR_DRACO_MESH_COMPRESSION ];
+		var geometryKey;
 
-		for ( var i = 0, il = a.length; i < il; i ++ ) {
+		if ( dracoExtension ) {
 
-			if ( a[ i ] !== b[ i ] ) return false;
+			geometryKey = 'draco:' + dracoExtension.bufferView
+				+ ':' + dracoExtension.indices
+				+ ':' + createAttributesKey( dracoExtension.attributes );
+
+		} else {
+
+			geometryKey = primitiveDef.indices + ':' + createAttributesKey( primitiveDef.attributes ) + ':' + primitiveDef.mode;
 
 		}
 
-		return true;
+		return geometryKey;
 
 	}
 
-	function getCachedGeometry( cache, newPrimitive ) {
+	function createAttributesKey( attributes ) {
+
+		var attributesKey = '';
 
-		for ( var i = 0, il = cache.length; i < il; i ++ ) {
+		var keys = Object.keys( attributes ).sort();
 
-			var cached = cache[ i ];
+		for ( var i = 0, il = keys.length; i < il; i ++ ) {
 
-			if ( isPrimitiveEqual( cached.primitive, newPrimitive ) ) return cached.promise;
+			attributesKey += keys[ i ] + ':' + attributes[ keys[ i ] ] + ';';
 
 		}
 
-		return null;
+		return attributesKey;
 
 	}
 
-	function getCachedCombinedGeometry( cache, geometries ) {
+	function createArrayKeyBufferGeometry( a ) {
 
-		for ( var i = 0, il = cache.length; i < il; i ++ ) {
+		var arrayKey = '';
 
-			var cached = cache[ i ];
+		for ( var i = 0, il = a.length; i < il; i ++ ) {
 
-			if ( isArrayEqual( geometries, cached.baseGeometries ) ) return cached.geometry;
+			arrayKey += i + a[ i ].uuid;
 
 		}
 
-		return null;
+		return arrayKey;
 
 	}
 
-	function getCachedMultiPassGeometry( cache, geometry, primitives ) {
+	function createArrayKeyGLTFPrimitive( a ) {
 
-		for ( var i = 0, il = cache.length; i < il; i ++ ) {
+		var arrayKey = '';
 
-			var cached = cache[ i ];
+		for ( var i = 0, il = a.length; i < il; i ++ ) {
 
-			if ( geometry === cached.baseGeometry && isArrayEqual( primitives, cached.primitives ) ) return cached.geometry;
+			arrayKey += i + createGeometryKey( a[ i ] );
 
 		}
 
-		return null;
+		return arrayKey;
+
+	}
+
+	function createMultiPassGeometryKey( geometry, primitives ) {
+
+		return createGeometryKey( geometry ) + createArrayKeyGLTFPrimitive( primitives );
```

Comment:
Because the `createArrayKeyGLTFPrimitive` function is only used in one place, I think we can just roll its for loop into this function. Also, `createGeometryKey` expects a GLTF primitive as an argument, not a BufferGeometry. Just `geometry.uuid` should suffice here.